### PR TITLE
Add Windows CI using conda

### PR DESCRIPTION
**Pull request by czgdp1807**

### CODE REVIEW

**Comment by czgdp1807** [comment]
I guess `powershell` can be used on Github actions CI by setting `shell: powershell`. Now I have to run all the commands on Windows and then test on CI here.

File: .github/workflows/conda-win.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+name: Windows Conda End-to-end Test
```
Comment (timkpaine) [inline .github/workflows/conda-win.yml:1]: Please use the same file

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -173,7 +173,6 @@ jobs:
             - ubuntu-22.04 # https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md
             - macos-12 # https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md
             - macos-14 # https://github.com/actions/runner-images/blob/main/images/macos/macos-14-arm64-Readme.md
-            # - windows-2022 # https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md
```
Comment (timkpaine) [inline .github/workflows/build.yml:176]: leave all this, it will be needed

File: conda/dev-environment-win.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: csp-dev
+channels:
+  - conda-forge
+  - nodefaults
+dependencies:
+ - compilers
```
Comment (timkpaine) [inline conda/dev-environment-win.yml:6]: this is unsufficient and unsorted.

**Review by timkpaine — CHANGES_REQUESTED** [review]

File: .github/workflows/conda-win.yml (hunk 1 above)
Comment (czgdp1807) [reply .github/workflows/conda-win.yml:1]: I think Windows needs `powershell` and the other file uses `micromamba shell`. So for now I am using separate file. After resolving the issues locally I will combine the two files together.

File: conda/dev-environment-win.yml (hunk 3 above)
Comment (czgdp1807) [reply conda/dev-environment-win.yml:6]: Oh I see. I will see what else to install on Windows.

**Pull request closed without merging**